test/e2e/node/pod_resize.go

```diff
@@ -768,6 +769,201 @@ func doPodResizeRetryDeferredTests(f *framework.Framework) {
+
+		for i, pod := range testPods {
+			ginkgo.By(fmt.Sprintf("Create pod '%s' that fits the node '%s'", pod.Name, node.Name))
+			testPods[i] = podClient.CreateSync(ctx, pod)
+			gomega.Expect(testPods[i].Status.Phase).To(gomega.Equal(v1.PodRunning))
+			gomega.Expect(testPods[i].Generation).To(gomega.BeEquivalentTo(1))
+		}
+
+		testPod2CPUQuantityResized := resource.NewMilliQuantity(initNodeAvailableMilliCPU/3, resource.DecimalSI)
+		testPod2MemoryQuantityResized := resource.NewQuantity(initNodeAvailableMem/24, resource.DecimalSI)
+
+		testPod3CPUQuantityResized := resource.NewMilliQuantity(initNodeAvailableMilliCPU/24, resource.DecimalSI)
+		testPod3MemoryQuantityResized := resource.NewQuantity(2*initNodeAvailableMem/3, resource.DecimalSI)
+
+		testPod4CPUQuantityResized := resource.NewMilliQuantity(initNodeAvailableMilliCPU/2+initNodeAvailableMilliCPU/8, resource.DecimalSI)
+		testPod4MemoryQuantityResized := initialMemoryQuantity
+
+		patchTestpod2ToDeferred := fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod2CPUQuantityResized.MilliValue(), testPod2MemoryQuantityResized.Value())
+
+		patchTestpod3ToDeferred := fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod3CPUQuantityResized.MilliValue(), testPod3MemoryQuantityResized.Value())
+
+		patchTestpod4ToDeferred := fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod4CPUQuantityResized.MilliValue(), testPod4MemoryQuantityResized.Value())
+
+		patches := []string{patchTestpod2ToDeferred, patchTestpod3ToDeferred, patchTestpod4ToDeferred}
+
+		for i := range patches {
+			testPod := testPods[i+1]
+			patch := patches[i]
+			ginkgo.By(fmt.Sprintf("Resize pod '%s' that cannot fit node due to insufficient CPU or memory", testPod.Name))
+			testPod, err = f.ClientSet.CoreV1().Pods(testPod.Namespace).Patch(ctx,
+				testPod.Name, types.StrategicMergePatchType, []byte(patch), metav1.PatchOptions{}, "resize")
+			framework.ExpectNoError(err, "failed to patch pod for resize")
+			waitForPodDeferred(ctx, f, testPod)
+			gomega.Expect(testPod.Generation).To(gomega.BeEquivalentTo(2))
+		}
+
+		ginkgo.By("deleting pod 1")
+		delErr1 := e2epod.DeletePodWithWait(ctx, f.ClientSet, testPod1)
+		framework.ExpectNoError(delErr1, "failed to delete pod %s", testPod1.Name)
+
+		ginkgo.By(fmt.Sprintf("Verify pod '%s' is resized successfully after pod deletion '%s'", testPod2.Name, testPod1.Name))
+		expected := []podresize.ResizableContainerInfo{
```

Comment:
optional: if you move these expectation definitions up to where you define the patches (or maybe replace the definition of the resized resources), then you can use `podresize.MakeResizePatch(c, desired)` to generate the patches, and cut down on some duplication.

Reply:
done